Source/utils/display.cpp

```diff
@@ -135,12 +135,7 @@ void AdjustToScreenGeometry(Size windowSize)
 {
 	gnScreenWidth = windowSize.width;
 	gnScreenHeight = windowSize.height;
-
-	gnViewportHeight = gnScreenHeight;
-	if (gnScreenWidth <= PANEL_WIDTH) {
-		// Part of the screen is fully obscured by the UI
-		gnViewportHeight -= PANEL_HEIGHT;
-	}

```

Comment:
This bit of code was moved to `CalculatePanelAreas()` but you are calling `CalculateUIRectangle()` in it's replacment, is that intentional?

Reply:
I added it, cause if we want to support changing resolution while ingame it would make a difference.
Normally `InitControlPan` is called when entering a game, so this should make it always correct.
The tests still need to call it, cause the tests also don't call `AdjustToScreenGeometry` and before used the on-the-fly calculation.